server/src/utils/callAutomationUtils.ts

```diff
@@ -0,0 +1,294 @@
+  return await callConnection.getCallMedia().stopTranscription();
+};
+
+/**
+ * We should be storing this based on the connectionId and not the correlationId
+ * We should re-write all the fetches based on the callId to use the serverCallId. This is
+ * because all of the clients that join can have different callId's and we need to make sure we are
+ * pulling the correct transcription data.
+ */
+export const TRANSCRIPTION_STORE: { [key: string]: Partial<CallTranscription> } = {};
+/**
+ * Used to map between the call connection id and the correlation id from both transcription and
+ * call automation events.
+ */
+export const CALLCONNECTION_ID_TO_CORRELATION_ID: {
+  [key: string]: { correlationId?: string; serverCallId: string };
+} = {};
+
+/**
+ * used to store the remote participants in the call
+ * This object is keyed off the callId and contains the communicationUserId and displayName of the remote participants
+ *
+ * Keeps track of all participants who have ever joined the call so we can show their display name in the transcription and summary.
+ */
+export const REMOTE_PARTICIPANTS_IN_CALL: {
+  [key: string]: { communicationUserId: string; displayName: string }[];
+} = {};
+
+/**
+ * used to store the local participant in the call
+ * This is keyed off the callId and contains the communicationUserId and displayName of the local participant
+ *
+ * This is important because the server can track the user across multiple calls if they change their display name so we can show
+ * the correct display name in the transcription and summary.
+ */
+export const LOCAL_PARTICIPANT: { [key: string]: { communicationUserId?: string; displayName?: string } } = {};
+
+export const getTranscriptionData = (serverCallId: string): CallTranscription | undefined => {
+  console.log('Getting transcription data for call:', serverCallId);
+  const connectionId = Object.keys(CALLCONNECTION_ID_TO_CORRELATION_ID).find((key) =>
+    CALLCONNECTION_ID_TO_CORRELATION_ID[key].serverCallId.includes(serverCallId)
+  );
+  if (!connectionId) {
+    return undefined;
+  }
+  const correlationId = CALLCONNECTION_ID_TO_CORRELATION_ID[connectionId]?.correlationId;
+  if (!correlationId) {
+    return undefined;
+  }
+  return TRANSCRIPTION_STORE[correlationId] as CallTranscription;
+};
+
+/**
+ * Check if transcription has started for the call
+ */
+export const checkIfTranscriptionStarted = (serverCallId: string): boolean => {
+  console.log('Checking if transcription started for call:', serverCallId);
+  const connectionId = Object.keys(CALLCONNECTION_ID_TO_CORRELATION_ID).find((key) =>
+    CALLCONNECTION_ID_TO_CORRELATION_ID[key].serverCallId.includes(serverCallId)
+  );
+  if (!connectionId) {
+    return false;
+  }
+  const correlationId = CALLCONNECTION_ID_TO_CORRELATION_ID[connectionId]?.correlationId;
+  return correlationId ? !!TRANSCRIPTION_STORE[correlationId] : false;
```

Comment:
something that a manager class would help clean up, instead of
```
  const correlationId = CALLCONNECTION_ID_TO_CORRELATION_ID[connectionId]?.correlationId;
  return correlationId ? !!TRANSCRIPTION_STORE[correlationId] : false;
```
we could have
```
return !!TranscriptionManager.getTranscriptions(correlationId);
```